SO-5498: Add support for "Simple map to SNOMED CT" reference sets

apeteri opened the pull request.

apeteri committed:
SO-5498: Remove generated EMF code from SnomedRefSetType; add literal...

...for "Simple map to SNOMED CT reference set" type

apeteri committed:
SO-5498: Add RF2 field name and header for "map to" reference sets

apeteri committed:
SO-5498: Add identifier concept parent SCTID for "map to" reference sets

apeteri committed:
SO-5498: Extend index entry with map source field; add entry builder

apeteri committed:
SO-5498: Add create and update delegates for "map to" reference sets

apeteri committed:
feat(snomed): Add "SIMPLE_MAP_TO" constant to MapTargetTypes; add...

...new constant to "supportedRefSetTypes" array on SNOMED CT concept and
description components

apeteri committed:
refactor(snomed): Rename parent concept constant in SnomedConstants

This reflects a PT change in the dataset -- the previous "Simple map"
parent should only contain "Simple map from SNOMED CT" reference sets.

apeteri committed:
feat(snomed): Add RF2 content type for "map to"-type reference sets

apeteri committed:
feat(snomed): Add "mapSourceComponentType" field to concept document

apeteri committed:
feat(snomed): Support "map to" reference sets in RF2 transaction context

- Prefix map target and source component types with the tooling ID
- Populate mapSourceComponentType for reference sets of the new type
- Populate mapSourceId for members of the new reference set type

apeteri committed:
feat(snomed): Support "map to" reference sets in SnomedRefSetUtil

apeteri committed:
feat(snomed): Move allowed referenced component types for simple...

...map reference sets in RefSetSupport; add entries for the new "map to"
type

apeteri committed:
chore(snomed): Add javadoc to SnomedRefSetType

apeteri committed:
refactor(snomed): Simplify "changed" flag handling in...

...SnomedSimpleMapToMemberUpdateDelegate as there is only a single
property to be updated.

apeteri committed:
feat(snomed): Add "mapSourceComponentType" to SnomedReferenceSet and...

...populate the value in the corresponding converter class

apeteri committed:
feat(snomed): Enable setting "mapSourceComponentType" in reference...

...set create requests

apeteri committed:
feat(snomed): Support "position-independent" search of map sources...

...and targets in SnomedRefSetMemberSearchRequest

apeteri committed:
chore(snomed): Consolidate String emptiness-checking methods

apeteri committed:
feat(core): Update method name for "source IDs" filter when searching...

...for mappings in concept maps (ConceptMapMappingSearchRequest),
allowing to handle results in "map to" type reference sets in a
transparent manner.

apeteri committed:
feat(snomed): Add support for "map to"-type reference sets in SNOMED...

...CT-specific concept map evaluator

apeteri committed:
feat(snomed): Add RF2 export support for "map to"-type reference sets

apeteri committed:
test(snomed): Check that populated "map source component type" field...

...is returned correctly from the index in
SnomedConceptDocumentSerializationTest

apeteri committed:
feat(snomed): Connect create and update delegates for "map to" type...

...in member create and update requests

apeteri committed:
test(snomed): Include "map to" type in reference set and member tests

apeteri committed:
test(snomed): Add valid property map for "map to" reference set members

apeteri committed:
test(snomed): Add import and export test cases for "map to"-type refsets

apeteri committed:
Merge branch '8.x' into feature/SO-5498_map_to_reference_sets

apeteri committed:
feat(snomed): Add constant for "Map source" reference set attribute

apeteri committed:
feat(snomed): Update reference set descriptor generation logic

- "referencedComponentId" (attribute order 0) becomes "Map source" or
"Map target" in the descriptor for mapping type reference sets
- "mapTarget" (attribute order 1) is now always registered as a "Map
target" attribute
- added "mapSource" (attribute order 1) as a "Map source" attribute for
the new reference set type

cmark reviewed (approved):
This looks really great, thanks @apeteri!

apeteri committed:
feat(snomed): Add icon for "map to" reference set type

apeteri committed:
feat(snomed): Support map source component type updates in request

apeteri committed:
fix(snomed): Add "should" clause to SnomedRefSetMemberSearchRequest...

...for OptionKey.COMPONENT; the provided component IDs should be
returned if they are referenced components, map sources or map targets.
